tslearn/preprocessing/preprocessing.py

```diff
@@ -296,3 +296,57 @@ def transform(self, X, y=None, **kwargs):
 
     def _more_tags(self):
         return {'allow_nan': True}
+
+
+class TimeSeriesScaleMeanMaxVariance(TimeSeriesScalerMeanVariance):
+    """Scaler for time series. Scales time series so that their mean (resp.
+    standard deviation) in the signal with the max amplitue  is
+    mu (resp. std). The scaling relationships between each signal are preserved
+    This is supplement to the TimeSeriesScalerMeanVariance method
+
+    Parameters
+    ----------
+    mu : float (default: 0.)
+        Mean of the output time series.
+    std : float (default: 1.)
+        Standard deviation of the output time series.
+
+    Notes
+    -----
+        This method requires a dataset of equal-sized time series.
+
+        NaNs within a time series are ignored when calculating mu and std.
+    """
+
+    def transform(self, X, y=None, **kwargs):
+        """Fit to data, then transform it.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_ts, sz, d)
+            Time series dataset to be rescaled
+
+        Returns
+        -------
+        numpy.ndarray
+            Rescaled time series dataset
+        """
+        check_is_fitted(self, '_X_fit_dims')
+        X = check_array(X, allow_nd=True, force_all_finite=False)
+        X_ = to_time_series_dataset(X)
+        X_ = check_dims(X_, X_fit_dims=self._X_fit_dims, extend=False)
+        mean_t = numpy.nanmean(X_, axis=1, keepdims=True)
+        std_t = numpy.nanstd(X_, axis=1, keepdims=True)
+        # retain the scaling relation cross the signals,
```

Comment:
Are these comments still relevant? Does it need to be stateful? I.e. should there be a fit() and transform() where in the fit() the self.std is stored and used in the transform() method?